Document password hashing configuration (#2826)
influxdata/docs-v2: content/enterprise_influxdb/v1.9/administration/configure-password-hashing.md

---
title: Configure password hashing
description: >
  Configure the cryptographic algorithm used for password hashing.
menu:
  enterprise_influxdb_1_9:
    name: Configure password hashing
    weight: 80
    parent: Administration
---

Compliance standards might require particular hashing alorithms
when storing the InfluxDB Enterprise [user](/enterprise_influxdb/v1.9/concepts/glossary/#user) passwords.

By default, InfluxDB Enterprise uses `bcrypt` for password hashing.
Use the following configuration options to configure the password hashing algorithm.

{{% note %}}
The `meta.password-hash` setting must be the same in both the data and meta node configuration files.
{{% /note %}}

## Example configuration

> **Example data node configuration:**

Author: Ah, that could be nice! I'll try it out.

```toml
[meta]
  # Configures password hashing scheme. Use "pbkdf2-sha256" or "pbkdf2-sha512"
  # for a FIPS-ready password hash. This setting must have the same value as
  # the meta nodes' meta.password-hash configuration.
  password-hash = "pbkdf2-sha256"

  # Configures strict FIPS-readiness check on startup.
  ensure-fips = true
```

**Example meta node configuration:**

```toml
[meta]
  # Configures password hashing scheme. Use "pbkdf2-sha256" or "pbkdf2-sha512"
  # for a FIPS-ready password hash. This setting must have the same value as
  # the data nodes' meta.password-hash configuration.
  password-hash = "pbkdf2-sha256"

  # Configures strict FIPS-readiness check on startup.
  ensure-fips = true
```

## Change password hashing algorithm

Complete the following steps
to change the password hashing algorithm used by an existing InfluxDB Enterprise cluster:

1. Ensure all meta and data nodes are running InfluxDB Enterprise 1.9.3 or later.

> 2. Edit the meta and data node configuration files as in the [example above](#example-configuration).

Reviewer: update if needed

---
title: Configure password hashing
description: >
  Configure the cryptographic algorithm used for password hashing.
menu:
  enterprise_influxdb_1_9:
    name: Configure password hashing
    weight: 80
    parent: Administration
related:
    - /enterprise_influxdb/v1.9/administration/configuration/
---

By default, InfluxDB Enterprise uses `bcrypt` for password hashing.

> [FIPS] compliance requires particular hashing alorithms.

Reviewer: @pierwill did you mean to add a link for FIPS?

Author: the link is in on line 65

> [FIPS] compliance standards require particular hashing alorithms.

Reviewer: Did you mean to add link for FIPS?

Author: Yep! It's at the bottom.

Use `pbkdf2-sha256` or `pbkdf2-sha256` for FIPS compliance.

## Change password hashing algorithm

Complete the following steps
to change the password hashing algorithm used by an existing InfluxDB Enterprise cluster:

1. Ensure all meta and data nodes are running InfluxDB Enterprise 1.9.3 or later.

> 2. In your meta node and data node configuration files, set [`password-hash`] to your desired algorithm

Reviewer: ```suggestion
2. In your meta node and data node configuration files, set [`password-hash`] to one of the following algorithms: `bcrypt` (default), `pbkdf2-sha256`, and `pbkdf2-sha512`
```

>    and [`ensure-fips`] to `true`.

Reviewer: ```suggestion
   and set [`ensure-fips`] to `true`.
```

   {{% note %}}
The `meta.password-hash` setting must be the same in both the data and meta node configuration files.
   {{% /note %}}

> 3. Restart each meta and node to pick up the configuration change.

Reviewer: Each meta and data node?

4. [Reset](/enterprise_influxdb/v1.9/administration/authentication_and_authorization/#reset-a-users-password) all user passwords.

>    For all existing usernames within the cluster,
>    the passwords *must* be reset in order for the new hashing algorithm to be applied.

Reviewer: ```suggestion
To apply the new hashing algorithm, you must reset all existing usernames in the cluster.
```

   Otherwise, the previous algorithm will continue to be used.

## Example configuration

**Example data node configuration:**

```toml
[meta]
  # Configures password hashing scheme. Use "pbkdf2-sha256" or "pbkdf2-sha512"
  # for a FIPS-ready password hash. This setting must have the same value as
  # the meta nodes' meta.password-hash configuration.
  password-hash = "pbkdf2-sha256"

  # Configures strict FIPS-readiness check on startup.
  ensure-fips = true
```

**Example meta node configuration:**

```toml
[meta]
  # Configures password hashing scheme. Use "pbkdf2-sha256" or "pbkdf2-sha512"
  # for a FIPS-ready password hash. This setting must have the same value as
  # the data nodes' meta.password-hash configuration.
  password-hash = "pbkdf2-sha256"

  # Configures strict FIPS-readiness check on startup.
  ensure-fips = true
```

[FIPS]: https://csrc.nist.gov/publications/detail/fips/140/3/final
[`password-hash`]: /enterprise_influxdb/v1.9/administration/config-meta-nodes/#password-hash--bcrypt

> [`ensure-fips`]: /enterprise_influxdb/v1.9/administration/config-meta-nodes/#ensure-fips--false

Reviewer: @pierwill did you mean to insert lines 65-67 as links elsewhere?

Author: those are reference style links. I'm playing with them for certain longer links since I personally like the style, but I can replace them if we don't want to use that style.

Reviewer: ah, I see, that works!

  # Configures strict FIPS-readiness check on startup.
  ensure-fips = true
```

## Change password hashing algorithm

To change the password hashing algorithm used by an existing InfluxDB Enterprise cluster:

1. Ensure all meta and data nodes are running InfluxDB Enterprise 1.9.3 or later.
2. Edit the meta and data node configuration files as in the [example above](#example-configuration).
2. Restart each meta and node to pick up the configuration change.

> 3. Change each password. This will store each password in the new hash format.

Reviewer 2: Numbering is 1, 2, 2, 3 ...perhaps 1,2,3,4?

This last instruction isn't a very clear.  Essentially, you are suggesting that for all existing usernames within the cluster, the passwords MUST be updated in order for the new hashing algorithm to be applied.  If the passwords are NOT updated, the old algorithm will continue to be used.

How is an administrator to know whether they have compliance?  Meaning, is there a way to check which algo is in use? Which users need prompting to make changes?
